libs/store/src/lib/infrastructure/persistence/store/entity-store-server/entity-store.ts

```diff
@@ -0,0 +1,94 @@
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { EntityCollectionService } from '@ngrx/data';
+
+import { CachePolicy, FetchPolicy, IdentityKey } from '../../../../domain/public_api';
+
+//#region Interfaces
+
+export type QueryParams = { [name: string]: string | string[] } | string;
+
+/**
+ * Entity Store interface
+ */
+export interface EntityStore<T> {
+    /**
+     * Query builder reference.
+     * Use it to create a query to underlying entity collection
+     */
+    readonly queryBuilder: any;
+    /**
+     * Get collection of entities
+     */
+    getAll(): Observable<T[]>;
+
+    /**
+     * Get collection of entities that match given query
+     */
+    getWithQuery(query: QueryParams): Observable<T[]>;
```

Comment:
getWithQuery should be internal API that is used by Query, but user will not use this really

Reply:
removed